--- ./src/xitk/Imlib-light/load.c~	2010-03-02 02:02:05.000000000 +0900
+++ ./src/xitk/Imlib-light/load.c	2014-12-05 22:53:06.150760648 +0900
@@ -61,7 +61,7 @@
       png_destroy_read_struct(&png_ptr, NULL, NULL);
       return NULL;
     }
-  if (info_ptr->color_type == PNG_COLOR_TYPE_RGB_ALPHA)
+  if (png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_RGB_ALPHA)
     {
       png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
       return NULL;
