--- ./tmac/troffrc.orig	2010-03-23 16:08:26.000000000 +0900
+++ ./tmac/troffrc	2010-03-23 16:23:29.000000000 +0900
@@ -48,4 +48,12 @@
 .el .ie "\V[LANG]"zh_SG.GBK"   .do mso gb.tmac
 .el .if "\V[LANG]"zh_TW"       .do mso big5.tmac
 .
+.\" convert hypyen/minus variants to ASCII-compatible HYPHEN-MINUS
+.if '\*[.T]'utf8' \{\
+.    char - \N'45'
+.    char \- \N'45'
+.    char ' \N'39'
+.    char \' \N'39'
+.\}
+
 .\" Don't let blank lines creep in here.
