From 302665451469e50631e3881a1c9a8d54f0f87ca7 Mon Sep 17 00:00:00 2001
From: KATOH Yasufumi <karma@jazz.email.ne.jp>
Date: Mon, 7 Feb 2022 15:14:11 +0900
Subject: [PATCH] prevent raw.8 missing error

---
 disk-utils/Makemodule.am | 2 --
 1 file changed, 2 deletions(-)

diff --git a/disk-utils/Makemodule.am b/disk-utils/Makemodule.am
index 2124612..5be93d8 100644
--- a/disk-utils/Makemodule.am
+++ b/disk-utils/Makemodule.am
@@ -110,8 +110,6 @@ endif
 
 if BUILD_RAW
 sbin_PROGRAMS += raw
-MANPAGES += disk-utils/raw.8
-dist_noinst_DATA += disk-utils/raw.8.adoc
 raw_SOURCES = disk-utils/raw.c
 endif
 
-- 
2.35.0

