diff -uNr cyrus-sasl-2.1.23.orig/Makefile.am cyrus-sasl-2.1.23/Makefile.am
--- cyrus-sasl-2.1.23.orig/Makefile.am	2006-05-19 23:45:21.000000000 +0900
+++ cyrus-sasl-2.1.23/Makefile.am	2014-02-28 18:54:46.620261090 +0900
@@ -79,7 +79,7 @@
 
 framedir = /Library/Frameworks/SASL2.framework
 install-exec-local: $(INSTALLOSX)
-	@if test "$(plugindir)" != "$(prefix)/lib/sasl2"; then \
+	@if test "$(plugindir)" != "$(libdir)/sasl2"; then \
 	  echo "********************************************************"; \
 	  echo "* WARNING:"; \
 	  echo "* Plugins are being installed into $(prefix)/lib/sasl2,"; \
diff -uNr cyrus-sasl-2.1.23.orig/Makefile.in cyrus-sasl-2.1.23/Makefile.in
--- cyrus-sasl-2.1.23.orig/Makefile.in	2009-05-07 23:24:44.000000000 +0900
+++ cyrus-sasl-2.1.23/Makefile.in	2014-02-28 18:54:32.082983026 +0900
@@ -672,7 +672,7 @@
 	@find $(distdir) -name CVS -print | xargs -t rm -rf
 	(cd $(distdir)/plugins && sh makeinit.sh)
 install-exec-local: $(INSTALLOSX)
-	@if test "$(plugindir)" != "$(prefix)/lib/sasl2"; then \
+	@if test "$(plugindir)" != "$(libdir)/sasl2"; then \
 	  echo "********************************************************"; \
 	  echo "* WARNING:"; \
 	  echo "* Plugins are being installed into $(prefix)/lib/sasl2,"; \
diff -uNr cyrus-sasl-2.1.23.orig/plugins/Makefile.am cyrus-sasl-2.1.23/plugins/Makefile.am
--- cyrus-sasl-2.1.23.orig/plugins/Makefile.am	2009-04-29 00:09:17.000000000 +0900
+++ cyrus-sasl-2.1.23/plugins/Makefile.am	2014-02-28 18:54:46.620261090 +0900
@@ -75,7 +75,7 @@
 
 common_sources = plugin_common.c plugin_common.h
 
-sasldir = $(prefix)/lib/sasl2
+sasldir = $(plugindir)/sasl2
 sasl_LTLIBRARIES = @SASL_MECHS@
 EXTRA_LTLIBRARIES = libplain.la libanonymous.la libkerberos4.la libcrammd5.la \
 	libgssapiv2.la libdigestmd5.la liblogin.la libsrp.la libotp.la \
diff -uNr cyrus-sasl-2.1.23.orig/plugins/Makefile.in cyrus-sasl-2.1.23/plugins/Makefile.in
--- cyrus-sasl-2.1.23.orig/plugins/Makefile.in	2009-05-07 23:24:56.000000000 +0900
+++ cyrus-sasl-2.1.23/plugins/Makefile.in	2014-02-28 18:55:05.271044712 +0900
@@ -263,7 +263,7 @@
 
 common_sources = plugin_common.c plugin_common.h
 
-sasldir = $(prefix)/lib/sasl2
+sasldir = $(plugindir)/sasl2
 sasl_LTLIBRARIES = @SASL_MECHS@
 EXTRA_LTLIBRARIES = libplain.la libanonymous.la libkerberos4.la libcrammd5.la \
 	libgssapiv2.la libdigestmd5.la liblogin.la libsrp.la libotp.la \
