diff -uNr dcron-4.5/extra/crond.rc build/extra/crond.rc
--- dcron-4.5/extra/crond.rc	2011-05-01 20:46:17.000000000 +0900
+++ build/extra/crond.rc	2017-01-18 17:00:47.299950074 +0900
@@ -1,36 +1,35 @@
 #!/bin/bash
 
-. /etc/rc.conf
-. /etc/rc.d/functions
-. /etc/conf.d/crond
+#. /etc/rc.conf
+#. /etc/rc.d/functions
+#. /etc/conf.d/crond
 
 PID=$(pidof -o %PPID /usr/sbin/crond)
 case $1 in
 start)
-	stat_busy "Starting Cron Daemon"
+	echo "Starting Cron Daemon"
 
 	# defaults to using syslog, and sendmail-ing cron output to local user
 	# to mail output to remote address instead, add "-m user@host"
 	# to CROND_ARGS in /etc/conf.d/crond
-	if [[ -z $PID ]] && env -i PATH="/sbin:/usr/sbin:/bin:/usr/bin" /usr/sbin/crond $CROND_ARGS; then
+	if [[ -z $PID ]] && env -i PATH="/sbin:/usr/sbin:/bin:/usr/bin" /usr/sbin/crond -S -l info; then
 
 		PID=$(pidof -o %PPID /usr/sbin/crond)
 		echo "$PID" > /var/run/crond.pid
-		add_daemon crond
-		stat_done
+		echo "crond started"
 	else
-		stat_fail
+		echo "crond failed to start"
 		exit 1
 	fi
 	;;
 
 stop)
-	stat_busy "Stopping Cron Daemon"
+	echo "Stopping Cron Daemon"
 	if [[ ! -z $PID ]]  && kill "$PID" &>/dev/null; then
-		rm_daemon crond
-		stat_done
+		rm -f /var/run/crond.pid
+		echo "crond stopped"
 	else
-		stat_fail
+		echo "crond failed to stop"
 		exit 1
 	fi
 	;;
