diff -upr nfs-utils-1.2.8/aclocal/kerberos5.m4 Temp/aclocal/kerberos5.m4
--- nfs-utils-1.2.8/aclocal/kerberos5.m4	2013-05-02 21:21:12.000000000 +0900
+++ Temp/aclocal/kerberos5.m4	2013-10-09 08:57:28.992213456 +0900
@@ -58,7 +58,11 @@ AC_DEFUN([AC_KERBEROS_V5],[
       dnl of Heimdal Kerberos on SuSe
       elif test \( -f $dir/include/heim_err.h -o\
       		 -f $dir/include/heimdal/heim_err.h \) -a \
-                -f $dir/lib/libroken.a; then
+                \( -f $dir/lib/libroken.a -o \
+                   -f $dir/lib64/libroken.a -o \
+                   -f $dir/lib64/libroken.so -o \
+                   -f $dir/lib/libroken.so \) ; then
+         K5VERS=`$K5CONFIG --version | head -n 1 | awk '{split($(2),v,"."); if (v@<:@"3"@:>@ == "") v@<:@"3"@:>@ = "0"; print v@<:@"1"@:>@v@<:@"2"@:>@v@<:@"3"@:>@ }'`
          AC_DEFINE(HAVE_HEIMDAL, 1, [Define this if you have Heimdal Kerberos libraries])
          KRBDIR="$dir"
          gssapi_lib=gssapi
diff -upr nfs-utils-1.2.8/configure Temp/configure
--- nfs-utils-1.2.8/configure	2013-05-03 03:20:24.000000000 +0900
+++ Temp/configure	2013-10-09 08:59:32.062413527 +0900
@@ -18719,7 +18719,7 @@ $as_echo "#define USE_GSS_KRB5_CCACHE_NA
          break
                   elif test \( -f $dir/include/heim_err.h -o\
       		 -f $dir/include/heimdal/heim_err.h \) -a \
-                -f $dir/lib/libroken.a; then
+                \( -f $dir/lib/libroken.so -o -f $dir/lib64/libroken.so \); then
 
 $as_echo "#define HAVE_HEIMDAL 1" >>confdefs.h
 
diff -upr nfs-utils-1.2.8/utils/gssd/krb5_util.c Temp/utils/gssd/krb5_util.c
--- nfs-utils-1.2.8/utils/gssd/krb5_util.c	2013-05-02 21:21:12.000000000 +0900
+++ Temp/utils/gssd/krb5_util.c	2013-10-09 08:57:28.995546794 +0900
@@ -958,9 +958,37 @@ check_for_tgt(krb5_context context, krb5
 {
 	krb5_error_code ret;
 	krb5_creds creds;
-	krb5_cc_cursor cur;
 	int found = 0;
 
+#ifdef HAVE_HEIMDAL
+	krb5_creds pattern;
+	krb5_const_realm client_realm;
+
+	krb5_cc_clear_mcred(&pattern);
+
+	client_realm = krb5_principal_get_realm (context, principal);
+
+	ret = krb5_make_principal (context, &pattern.server,
+				   client_realm, KRB5_TGS_NAME, client_realm,
+				   NULL);
+	if (ret)
+	  krb5_err (context, 1, ret, "krb5_make_principal");
+	pattern.client = principal;
+
+	ret = krb5_cc_retrieve_cred (context, ccache, 0, &pattern, &creds);
+	krb5_free_principal (context, pattern.server);
+	if (ret) {
+	  if (ret == KRB5_CC_END)
+            return 1;
+	  krb5_err (context, 1, ret, "krb5_cc_retrieve_cred");
+	}
+
+	found = creds.times.endtime > time(NULL);
+
+	krb5_free_cred_contents (context, &creds);
+#else
+	krb5_cc_cursor cur;
+
 	ret = krb5_cc_start_seq_get(context, ccache, &cur);
 	if (ret) 
 		return 0;
@@ -980,6 +1008,7 @@ check_for_tgt(krb5_context context, krb5
 		krb5_free_cred_contents(context, &creds);
 	}
 	krb5_cc_end_seq_get(context, ccache, &cur);
+#endif
 
 	return found;
 }
@@ -1026,6 +1055,9 @@ query_krb5_ccache(const char* cred_cache
 	}
 	krb5_free_principal(context, principal);
 err_princ:
+#ifdef HAVE_HEIMDAL
+#define KRB5_TC_OPENCLOSE              0x00000001
+#endif
 	krb5_cc_set_flags(context, ccache,  KRB5_TC_OPENCLOSE);
 	krb5_cc_close(context, ccache);
 err_cache:
