=== modified file 'ibus-el-agent'
--- a/ibus-el-agent	2012-02-14 08:50:43 +0000
+++ b/ibus-el-agent	2012-02-17 02:52:57 +0000
@@ -136,6 +136,13 @@
         self.top = 0
         self.focus = None
         self.focus_cb_id = None
+        display.set_error_handler(self.__error_handler)
+
+    # When the active window vanishes, focus.query_tree() etc. may cause
+    # BadWIndow error unwantedly.  This handler quiets all Xlib's errors,
+    # but we still can check them in "*ibus-mode log*" buffer if necessary.
+    def __error_handler(self, error, request):
+        print '("%s")'%("%s"%error).replace('"', '\\"')
 
     def __update_focus_cb(self, form, sync=False):
         # Check FocusIn/FocusOut events

