--- libtta-c-2.0/console/tta.h	2010-12-28 21:37:19.000000000 +0900
+++ build/console/tta.h	2011-01-29 19:00:40.000000000 +0900
@@ -22,6 +22,7 @@
 #endif
 #include <sys/types.h>
 #include <sys/time.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <stdio.h>
 #include <locale.h>
@@ -54,7 +55,7 @@
 typedef int (HANDLE);
 #define INVALID_HANDLE_VALUE (-1)
 #define tta_open_read(__name) open(__name,O_RDONLY|O_NONBLOCK)
-#define tta_open_write(__name) open(__name,O_WRONLY|O_TRUNC|O_CREAT)
+#define tta_open_write(__name) open(__name,O_WRONLY|O_TRUNC|O_CREAT, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH)
 #define tta_close(__handle) close(__handle)
 #define tta_unlink(__name) unlink(__name)
 #define tta_read(__handle,__buffer,__size,__result) (__result=read(__handle,__buffer,__size))
