--- build32/applets/wncklet/workspace-switcher.c.orig	2014-09-20 23:32:20.000000000 +0900
+++ build32/applets/wncklet/workspace-switcher.c	2014-12-26 09:19:35.000000000 +0900
@@ -143,7 +143,7 @@
 
 	if (!wm_name)
 		pager->wm = PAGER_WM_UNKNOWN;
-	else if (strcmp(wm_name, "Marco") == 0)
+	else if (strcmp(wm_name, "Metacity (Marco)") == 0)
 		pager->wm = PAGER_WM_MARCO;
 	else if (strcmp(wm_name, "Compiz") == 0)
 		pager->wm = PAGER_WM_COMPIZ;
