Signed-off-by: Etienne Le Sueur <elesueur@vmware.com>
diff --git a/sysdeps/gnu/configure b/sysdeps/gnu/configure
index 634e1d0..7f26ba4 100644
--- a/sysdeps/gnu/configure
+++ b/sysdeps/gnu/configure
@@ -17,8 +17,6 @@ case "$prefix" in
     libc_cv_slibdir=/lib64
     if test "$libdir" = '${exec_prefix}/lib'; then
       libdir='${exec_prefix}/lib64';
-      # Locale data can be shared between 32bit and 64bit libraries
-      libc_cv_localedir='${exec_prefix}/lib/locale'
     fi
     ;;
   *)
@@ -31,6 +29,12 @@ case "$prefix" in
   else
     libc_cv_sysconfdir=$sysconfdir
    fi
+
+  # Allow the user to override the path to locale data  with --localedir.
+  # $localedir is set regardless of --localedir configure argument,
+  # so we just set libc_cv_localedir here as well.
+  # NB: Locale data can be shared between 32bit and 64bit libraries
+  libc_cv_localedir=$localedir
   libc_cv_rootsbindir=/sbin
   ;;
 esac
diff --git a/sysdeps/gnu/configure.in b/sysdeps/gnu/configure.in
index 3422007..0caafeb 100644
--- a/sysdeps/gnu/configure.in
+++ b/sysdeps/gnu/configure.in
@@ -17,8 +17,6 @@ case "$prefix" in
     libc_cv_slibdir=/lib64
     if test "$libdir" = '${exec_prefix}/lib'; then
       libdir='${exec_prefix}/lib64';
-      # Locale data can be shared between 32bit and 64bit libraries
-      libc_cv_localedir='${exec_prefix}/lib/locale'
     fi
     ;;
   *)
@@ -31,6 +29,13 @@ case "$prefix" in
   else
     libc_cv_sysconfdir=$sysconfdir
    fi
+
+  # Allow the user to override the path to locale data with --localedir.
+  # $localedir is set regardless of --localedir configure argument,
+  # so we just set libc_cv_localedir here as well.
+  # NB: Locale data can be shared between 32bit and 64bit libraries
+  libc_cv_localedir=$localedir
+
   libc_cv_rootsbindir=/sbin
   ;;
 esac
