--- vobcopy-1.1.2/Makefile	2008-10-08 07:18:12.000000000 +0900
+++ build/Makefile	2011-02-03 20:49:30.000000000 +0900
@@ -6,12 +6,12 @@
 #PREFIX += /usr/local
 #BINDIR = ${PREFIX}/bin
 #MANDIR = ${PREFIX}/man
-PREFIX += /usr/local
+PREFIX += /usr
 BINDIR = ${PREFIX}/bin
-MANDIR = ${PREFIX}/man
+MANDIR = ${PREFIX}/share/man
 LFS    = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CFLAGS += -I/usr//include
-LDFLAGS += -ldvdread -L/usr//lib
+CFLAGS += -I/usr/include
+LDFLAGS += -ldvdread -L/usr/${libdir}
 
 #This specifies the conversion from .c to .o 
 .c.o:
@@ -33,10 +33,8 @@
 	$(CC) -o vobcopy vobcopy.o dvd.o -ldvdread
 
 deb:
-        
 	echo "this here is really really experimental..."
 	dpkg-buildpackage -rfakeroot -us -uc -tc
-		
 
 clean :
 	rm -f vobcopy vobcopy.o dvd.o
@@ -48,12 +46,12 @@
 #	mkdir -p $(MANDIR)/man1
 #	cp vobcopy   $(BINDIR)/vobcopy
 #	cp vobcopy.1 $(MANDIR)/man1/vobcopy.1
-	install -d -m 755 $(BINDIR)
-	install -d -m 755 $(MANDIR)/man1
-	install -d -m 755 $(MANDIR)/de/man1
-	install -m 755 vobcopy $(BINDIR)/vobcopy
-	install -m 644 vobcopy.1 $(MANDIR)/man1/vobcopy.1
-	install -m 644 vobcopy.1.de $(MANDIR)/de/man1/vobcopy.1
+	install -d -m 755 $(DESTDIR)$(BINDIR)
+	install -d -m 755 $(DESTDIR)$(MANDIR)/man1
+	install -d -m 755 $(DESTDIR)$(MANDIR)/de/man1
+	install -m 755 vobcopy $(DESTDIR)$(BINDIR)/vobcopy
+	install -m 644 vobcopy.1 $(DESTDIR)$(MANDIR)/man1/vobcopy.1
+	install -m 644 vobcopy.1.de $(DESTDIR)$(MANDIR)/de/man1/vobcopy.1
 
 uninstall:
 	rm -f $(BINDIR)/vobcopy
