From e3abf8729c5f553ecd9b6f999a8d3786cf055989 Mon Sep 17 00:00:00 2001
From: KATOH Yasufumi <karma@jazz.email.ne.jp>
Date: Fri, 29 Nov 2024 22:22:03 +0900
Subject: [PATCH] Build apache module as NTS

---
 sapi/apache2handler/config.m4 | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/sapi/apache2handler/config.m4 b/sapi/apache2handler/config.m4
index e335721f..320081e1 100644
--- a/sapi/apache2handler/config.m4
+++ b/sapi/apache2handler/config.m4
@@ -120,11 +120,6 @@ if test "$PHP_APXS2" != "no"; then
       -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1
     ])
 
-  AS_IF([$APXS_HTTPD -V 2>/dev/null | grep 'threaded:.*yes' >/dev/null 2>&1], [
-    APACHE_THREADED_MPM=yes
-    enable_zts=yes
-  ], [APACHE_THREADED_MPM=no])
-
 AC_CONFIG_COMMANDS([apache2handler], [AS_VAR_IF([enable_zts], [yes],,
   [AS_VAR_IF([APACHE_THREADED_MPM], [no],
     [AC_MSG_WARN([
-- 
2.45.1

