diff -uNr openjpeg-1.5.2.orig/doc/Makefile.am openjpeg-1.5.2/doc/Makefile.am
--- openjpeg-1.5.2.orig/doc/Makefile.am	2014-03-27 19:58:08.000000000 +0900
+++ openjpeg-1.5.2/doc/Makefile.am	2014-04-22 19:49:49.521644752 +0900
@@ -33,7 +33,7 @@
 
 endif
 
-docsdir = $(docdir)-$(MAJOR_NR).$(MINOR_NR)
+docsdir = $(docdir)-$(MAJOR_NR).$(MINOR_NR).$(MICRO_NR)
 docs_DATA = $(top_builddir)/LICENSE
 
 dist_man1_MANS = \
