--- ./util/grub.d/10_linux.in.orig	2015-01-01 11:39:57.000000000 +0900
+++ ./util/grub.d/10_linux.in	2015-01-21 13:05:19.600206404 +0900
@@ -31,7 +31,7 @@
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  OS="${GRUB_DISTRIBUTOR} Linux"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 fi

