--- grub-2.02/grub-core/fs/xfs.c~	2017-04-24 18:15:49.000000000 +0900
+++ grub-2.02/grub-core/fs/xfs.c	2019-01-12 13:18:01.486428339 +0900
@@ -79,9 +79,19 @@
 #define XFS_SB_FEAT_INCOMPAT_SPINODES   (1 << 1)        /* sparse inode chunks */
 #define XFS_SB_FEAT_INCOMPAT_META_UUID  (1 << 2)        /* metadata UUID */
 
-/* We do not currently verify metadata UUID so it is safe to read such filesystem */
-#define XFS_SB_FEAT_INCOMPAT_SUPPORTED \
+/*
+ * Directory entries with ftype are explicitly handled by grub code.
+ *
+ * We do not currently verify metadata UUID, so it is safe to read filesystems
+ * with the XFS_SB_FEAT_INCOMPAT_META_UUID feature.
+ *
+ * We do not currently read the inode btrees, so it is safe to read filesystems
+ * with the XFS_SB_FEAT_INCOMPAT_SPINODES feature.
+ */
+
+#define XFS_SB_FEAT_INCOMPAT_SUPPORTED		\
 	(XFS_SB_FEAT_INCOMPAT_FTYPE | \
+         XFS_SB_FEAT_INCOMPAT_SPINODES | \
 	 XFS_SB_FEAT_INCOMPAT_META_UUID)
 
 struct grub_xfs_sblock
