--- grub-2.06/grub-core/fs/xfs.c~	2021-06-02 00:20:20.000000000 +0900
+++ grub-2.06/grub-core/fs/xfs.c	2021-10-17 09:16:33.961618186 +0900
@@ -208,14 +208,15 @@
   grub_uint32_t nextents;
   grub_uint16_t unused3;
   grub_uint8_t fork_offset;
-  grub_uint8_t unused4[37];
+  grub_uint8_t unused4[17]; // Last member of inode v2
+  grub_uint8_t unused5[20]; // First member of inode v3
   grub_uint64_t flags2;
-  grub_uint8_t unused5[48];
+  grub_uint8_t unused6[48]; // Last member of inode v3
 } GRUB_PACKED;
 
 #define XFS_V3_INODE_SIZE	sizeof(struct grub_xfs_inode)
-/* Size of struct grub_xfs_inode until fork_offset (included). */
-#define XFS_V2_INODE_SIZE	(XFS_V3_INODE_SIZE - 92)
+/* Size of struct grub_xfs_inode v2, up to unused4 included */
+#define XFS_V2_INODE_SIZE      (XFS_V3_INODE_SIZE - 76)
 
 struct grub_xfs_dirblock_tail
 {
