From 4ea297738301ed305319d0569ad160a978e1cae1 Mon Sep 17 00:00:00 2001
From: KATOH Yasufumi <karma@jazz.email.ne.jp>
Date: Mon, 9 Jun 2025 18:59:39 +0900
Subject: [PATCH] No Savedir sort

---
 src/copy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/copy.c b/src/copy.c
index 22d9830..58f6801 100644
--- a/src/copy.c
+++ b/src/copy.c
@@ -832,7 +832,7 @@ copy_dir (char const *src_name_in, char const *dst_name_in,
   struct cp_options non_command_line_options = *x;
   bool ok = true;
 
-  name_space = savedir (src_name_in, SAVEDIR_SORT_FASTREAD);
+  name_space = savedir (src_name_in, SAVEDIR_SORT_NONE);
   if (name_space == nullptr)
     {
       /* This diagnostic is a bit vague because savedir can fail in
-- 
2.48.1

