From 801fe31a50ea5fd04ae7c8857297ec4d5727be0c Mon Sep 17 00:00:00 2001
From: Matias De lellis <mati86dl@gmail.com>
Date: Wed, 29 Aug 2012 15:08:07 -0300
Subject: [PATCH] Wait 5 second before mount devices.

---
 thunar-volman/tvm-block-device.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/thunar-volman/tvm-block-device.c b/thunar-volman/tvm-block-device.c
index 56969ed..5059cba 100644
--- a/thunar-volman/tvm-block-device.c
+++ b/thunar-volman/tvm-block-device.c
@@ -708,7 +708,7 @@ tvm_block_device_mount_finish (GVolume      *volume,
 
 
 
-static void
+static gboolean
 tvm_block_device_mount (TvmContext *context)
 {
   GMountOperation *mount_operation;
@@ -751,6 +751,7 @@ tvm_block_device_mount (TvmContext *context)
       /* finish processing the device */
       tvm_device_handler_finished (context);
     }
+    return FALSE;
 }
 
 
@@ -881,7 +882,7 @@ automount_disc:
               if (automount)
                 {
                   /* mount the CD/DVD and continue with inspecting its contents */
-                  tvm_block_device_mount (context);
+                  g_timeout_add_seconds(5, tvm_block_device_mount, context);
                 }
             }
           else
@@ -904,7 +905,7 @@ automount_disc:
       if (automount)
         {
           /* mount the partition and continue with inspecting its contents */
-          tvm_block_device_mount (context);
+          g_timeout_add_seconds(5, tvm_block_device_mount, context);
         }
       else
         {
-- 
1.7.11.4

