--- unrar/makefile.unix	2009-05-03 14:18:45.000000000 +0900
+++ build/makefile.unix	2011-01-29 13:24:01.000000000 +0900
@@ -6,8 +6,8 @@
 #
 
 # Linux using GCC
-#CXX=g++
-#CXXFLAGS=-O2
+CXX=g++
+CXXFLAGS=$(target) -O2 -fPIC
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 STRIP=strip
 DESTDIR=/usr
@@ -136,13 +136,14 @@
 	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
-			install unrar $(DESTDIR)/bin
+			install unrar $(DESTDIR)/usr/bin
 
 uninstall-unrar:
-			rm -f $(DESTDIR)/bin/unrar
+			rm -f $(DESTDIR)/usr/bin/unrar
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		mkdir -p $(DESTDIR)/$(libdir)
+		install libunrar.so $(DESTDIR)/$(libdir)
 
 uninstall-lib:
 		rm -f $(DESTDIR)/lib/libunrar.so
