diff -uNr freetype-2.5.3.orig/modules.cfg freetype-2.5.3/modules.cfg
--- freetype-2.5.3.orig/modules.cfg	2014-03-13 15:14:38.324616309 +0900
+++ freetype-2.5.3/modules.cfg	2014-03-13 15:15:42.057285153 +0900
@@ -110,7 +110,7 @@
 AUX_MODULES += cache
 
 # TrueType GX/AAT table validation.  Needs ftgxval.c below.
-# AUX_MODULES += gxvalid
+AUX_MODULES += gxvalid
 
 # Support for streams compressed with gzip (files with suffix .gz).
 #
@@ -129,7 +129,7 @@
 
 # OpenType table validation.  Needs ftotval.c below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
