diff -uNr freetype-2.5.5.orig/modules.cfg freetype-2.5.5/modules.cfg
--- freetype-2.5.5.orig/modules.cfg	2015-02-16 18:46:46.542061576 +0900
+++ freetype-2.5.5/modules.cfg	2015-02-16 18:48:34.810625415 +0900
@@ -110,7 +110,7 @@
 AUX_MODULES += cache
 
 # TrueType GX/AAT table validation.  Needs ftgxval.c below.
-# AUX_MODULES += gxvalid
+AUX_MODULES += gxvalid
 
 # Support for streams compressed with gzip (files with suffix .gz).
 #
@@ -129,7 +129,7 @@
 
 # OpenType table validation.  Needs ftotval.c below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
