diff -uNr freetype-2.5.3.orig/include/config/ftoption.h freetype-2.5.3/include/config/ftoption.h
--- freetype-2.5.3.orig/include/config/ftoption.h	2014-03-13 15:14:38.327949448 +0900
+++ freetype-2.5.3/include/config/ftoption.h	2014-03-13 15:18:07.933405782 +0900
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
