diff -uNr webkit-1.6.3.orig/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h webkit-1.6.3/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h
--- webkit-1.6.3.orig/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h	2012-01-31 10:20:05.000000000 +0900
+++ webkit-1.6.3/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h	2012-03-09 14:08:40.419803540 +0900
@@ -24,6 +24,7 @@
 
 #if ENABLE(GLIB_SUPPORT)
 
+#include <glib.h>
 #include <algorithm>
 #include <wtf/Assertions.h>
 #include <wtf/Noncopyable.h>
