diff --git a/data/oses/Makefile.am b/data/oses/Makefile.am
index e1fb6fb..48e1f94 100644
--- a/data/oses/Makefile.am
+++ b/data/oses/Makefile.am
@@ -16,6 +16,7 @@ database_in_files =	\
 	macos.xml.in    \
 	openbsd.xml.in  \
 	opensuse.xml.in \
+	plamo.xml.in	\
 	rhl.xml.in      \
 	rhel.xml.in     \
 	solaris.xml.in  \
diff --git a/data/oses/plamo.xml.in b/data/oses/plamo.xml.in
new file mode 100644
index 0000000..c8e2c0a
--- /dev/null
+++ b/data/oses/plamo.xml.in
@@ -0,0 +1,34 @@
+<libosinfo version="0.0.1">
+
+  <os id="http://repository.plamolinux.org/pub/linux/Plamo/Plamo-5.x/">
+    <short-id>plamo5.3.1</short-id>
+    <_name>Plamo 5.3.1</_name>
+    <version>5.3.1</version>
+    <_vendor>Plamo</_vendor>
+    <family>linux</family>
+    <distro>plamo</distro>
+    <release-date>2015-02-02</release-date>
+
+    <media arch="i686">
+      <url>http://repository.plamolinux.org/isos/plamo-5.3.1_x86_64_dvd.iso</url>
+      <iso>
+        <volume-id>PlamoDVD</volume-id>
+        <application-id>Plamo 5.3.1 DVD</application-id>
+        <publisher-id></publisher-id>
+      </iso>
+      <kernel>isolinux/vmlinuz</kernel>
+      <initrd>isolinux/initrd.gz</initrd>
+    </media>
+    <media arch="x86_64">
+      <url>http://repository.plamolinux.org/isos/plamo-5.3.1_x86_64_dvd.iso</url>
+      <iso>
+        <volume-id>PlamoDVD</volume-id>
+        <application-id>Plamo 5.3.1 DVD</application-id>
+        <publisher-id></publisher-id>
+      </iso>
+      <kernel>isolinux/vmlinuz</kernel>
+      <initrd>isolinux/initrd.gz</initrd>
+    </media>
+  </os>
+
+</libosinfo>
diff --git a/po/POTFILES.in b/po/POTFILES.in
index a1a551f..5344d84 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1,6 +1,7 @@
 data/oses/netware.xml.in
 data/oses/solaris.xml.in
 data/oses/dos.xml.in
+data/oses/plamo.xml.in
 data/oses/mageia.xml.in
 data/oses/mandrake.xml.in
 data/oses/mandriva.xml.in
