diff -upr volume_key-volume_key-0.3.12/python/python/Makefile.am volume_key-volume_key-0.3.12.patched/python/python/Makefile.am
--- volume_key-volume_key-0.3.12/python/python/Makefile.am	2018-10-08 20:24:58.000000000 +0900
+++ volume_key-volume_key-0.3.12.patched/python/python/Makefile.am	2023-04-09 21:43:20.965785053 +0900
@@ -17,7 +17,7 @@
 ## Process this file with automake to produce Makefile.in
 ACLOCAL_AMFLAGS = -I m4
 AM_CPPFLAGS = $(blkid_CFLAGS) $(glib_CFLAGS) $(GPGME_CFLAGS) \
-	$(libcryptsetup_CFLAGS) $(nss_CFLAGS)
+	$(PYTHON3_CFLAGS) $(libcryptsetup_CFLAGS) $(nss_CFLAGS)
 LOCALEDIR_CPPFLAGS = -DLOCALEDIR='"$(localedir)"'
 
 SWIG_FLAGS = -python
