diff -uNr shadow-4.4.orig/libmisc/myname.c shadow-4.4/libmisc/myname.c
--- shadow-4.4.orig/libmisc/myname.c	2014-09-01 23:36:40.000000000 +0900
+++ shadow-4.4/libmisc/myname.c	2017-02-27 16:25:38.094830090 +0900
@@ -44,7 +44,6 @@
 /*@null@*/ /*@only@*/struct passwd *get_my_pwent (void)
 {
 	struct passwd *pw;
-	const char *cp = getlogin ();
 	uid_t ruid = getuid ();
 
 	/*
@@ -57,12 +56,6 @@
 	 * XXX - when running from su, will return the current user (not
 	 * the original user, like getlogin() does).  Does this matter?
 	 */
-	if ((NULL != cp) && ('\0' != *cp)) {
-		pw = xgetpwnam (cp);
-		if ((NULL != pw) && (pw->pw_uid == ruid)) {
-			return pw;
-		}
-	}
 
 	return xgetpwuid (ruid);
 }
diff -uNr shadow-4.4.orig/src/useradd.c shadow-4.4/src/useradd.c
--- shadow-4.4.orig/src/useradd.c	2016-12-02 04:41:50.000000000 +0900
+++ shadow-4.4/src/useradd.c	2017-02-27 16:26:06.806222765 +0900
@@ -2027,6 +2027,8 @@
 	is_shadow_grp = sgr_file_present ();
 #endif
 
+       get_defaults ();
+
 	process_flags (argc, argv);
 
 #ifdef ENABLE_SUBIDS
@@ -2036,8 +2038,6 @@
 	    (!user_id || (user_id <= uid_max && user_id >= uid_min));
 #endif				/* ENABLE_SUBIDS */
 
-	get_defaults ();
-
 #ifdef ACCT_TOOLS_SETUID
 #ifdef USE_PAM
 	{
