diff -uNr shadow-4.4.orig/src/su.c shadow-4.4/src/su.c
--- shadow-4.4.orig/src/su.c	2016-12-02 04:41:50.000000000 +0900
+++ shadow-4.4/src/su.c	2017-02-27 16:20:04.375305917 +0900
@@ -373,8 +373,8 @@
 		              stderr);
 		(void) kill (-pid_child, caught);
 
-		snprintf (kill_msg, _(" ...killed.\n"));
-		snprintf (wait_msg, _(" ...waiting for child to terminate.\n"));
+		snprintf (kill_msg, 256, _(" ...killed.\n"));
+		snprintf (wait_msg, 256, _(" ...waiting for child to terminate.\n"));
 
 		(void) signal (SIGALRM, kill_child);
 		(void) alarm (2);
