--- shadow-4.2.1.orig/etc/login.defs	2014-05-09 19:20:28.000000000 +0900
+++ shadow-4.2.1/etc/login.defs	2016-07-19 19:27:59.456911275 +0900
@@ -14,7 +14,7 @@
 #
 # Enable logging and display of /var/log/faillog login(1) failure info.
 #
-FAILLOG_ENAB		yes
+#FAILLOG_ENAB		yes
 
 #
 # Enable display of unknown usernames when login(1) failures are recorded.
@@ -29,7 +29,7 @@
 #
 # Enable logging and display of /var/log/lastlog login(1) time info.
 #
-LASTLOG_ENAB		yes
+#LASTLOG_ENAB		yes
 
 #
 # Enable checking and display of mailbox status upon login.
@@ -37,22 +37,22 @@
 # Disable if the shell startup files already check for mail
 # ("mailx -e" or equivalent).
 #
-MAIL_CHECK_ENAB		yes
+#MAIL_CHECK_ENAB		yes
 
 #
 # Enable additional checks upon password changes.
 #
-OBSCURE_CHECKS_ENAB	yes
+#OBSCURE_CHECKS_ENAB	yes
 
 #
 # Enable checking of time restrictions specified in /etc/porttime.
 #
-PORTTIME_CHECKS_ENAB	yes
+#PORTTIME_CHECKS_ENAB	yes
 
 #
 # Enable setting of ulimit, umask, and niceness from passwd(5) gecos field.
 #
-QUOTAS_ENAB		yes
+#QUOTAS_ENAB		yes
 
 #
 # Enable "syslog" logging of su(1) activity - in addition to sulog file logging.
@@ -78,7 +78,7 @@
 # If defined, ":" delimited list of "message of the day" files to
 # be displayed upon login.
 #
-MOTD_FILE	/etc/motd
+#MOTD_FILE	/etc/motd
 #MOTD_FILE	/etc/motd:/usr/lib/news/news-motd
 
 #
@@ -96,14 +96,14 @@
 # If defined, login(1) failures will be logged here in a utmp format.
 # last(1), when invoked as lastb(1), will read /var/log/btmp, so...
 #
-FTMP_FILE	/var/log/btmp
+#FTMP_FILE	/var/log/btmp
 
 #
 # If defined, name of file whose presence will inhibit non-root
 # logins.  The content of this file should be a message indicating
 # why logins are inhibited.
 #
-NOLOGINS_FILE	/etc/nologin
+#NOLOGINS_FILE	/etc/nologin
 
 #
 # If defined, the command name to display when running "su -".  For
@@ -141,7 +141,7 @@
 # If defined, an HZ environment parameter spec.
 #
 # for Linux/x86
-ENV_HZ		HZ=100
+#ENV_HZ		HZ=100
 # For Linux/Alpha...
 #ENV_HZ		HZ=1024
 
@@ -202,7 +202,7 @@
 #
 PASS_MAX_DAYS	99999
 PASS_MIN_DAYS	0
-PASS_MIN_LEN	5
+#PASS_MIN_LEN	5
 PASS_WARN_AGE	7
 
 #
@@ -211,12 +211,12 @@
 # to uid 0 accounts.  If the group doesn't exist or is empty, no one
 # will be able to "su" to uid 0.
 #
-SU_WHEEL_ONLY	no
+#SU_WHEEL_ONLY	no
 
 #
 # If compiled with cracklib support, sets the path to the dictionaries
 #
-CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
+#CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
 
 #
 # Min/max values for automatic uid selection in useradd(8)
@@ -257,12 +257,12 @@
 #
 # Maximum number of attempts to change password if rejected (too easy)
 #
-PASS_CHANGE_TRIES	5
+#PASS_CHANGE_TRIES	5
 
 #
 # Warn about weak passwords (but still allow them) if you are root.
 #
-PASS_ALWAYS_WARN	yes
+#PASS_ALWAYS_WARN	yes
 
 #
 # Number of significant characters in the password for crypt().
@@ -274,7 +274,7 @@
 #
 # Require password before chfn(1)/chsh(1) can make any changes.
 #
-CHFN_AUTH		yes
+#CHFN_AUTH		yes
 
 #
 # Which fields may be changed by regular users using chfn(1) - use
@@ -356,7 +356,7 @@
 # If this file exists and is readable, login environment will be
 # read from it.  Every line should be in the form name=value.
 #
-ENVIRON_FILE	/etc/environment
+#ENVIRON_FILE	/etc/environment
 
 #
 # If defined, this command is run when removing a user.
