diff -Nrc files/etc/csh.login files32/etc/csh.login
*** files/etc/csh.login	2009-08-07 23:13:22.000000000 +0900
--- files32/etc/csh.login	2009-08-07 23:13:22.000000000 +0900
***************
*** 17,23 ****
  	setenv LANG ja_JP.eucJP
          setenv JSERVER localhost
          setenv TZ /etc/localtime
!         setenv PKG_CONFIG_PATH /usr/lib64/pkgconfig
  	if( $SHLVL == 1 ) then
  	  set path = ( /usr/local/bin /bin /usr/bin )
  	  set path = ( $path /usr/games/bin /usr/games )
--- 17,23 ----
  	setenv LANG ja_JP.eucJP
          setenv JSERVER localhost
          setenv TZ /etc/localtime
!         setenv PKG_CONFIG_PATH /usr/lib/pkgconfig
  	if( $SHLVL == 1 ) then
  	  set path = ( /usr/local/bin /bin /usr/bin )
  	  set path = ( $path /usr/games/bin /usr/games )
***************
*** 29,35 ****
  	endif
  	## for JAVA
  	#setenv JAVA_HOME /usr/local/j2sdk1.4.0
! 	#setenv CLASSPATH ".:$JAVA_HOME/lib64/tools.jar"
  	#if( $SHLVL == 1 ) then
  	#  set path = ( $path $JAVA_HOME/jre/bin $JAVA_HOME/bin )
  	#endif
--- 29,35 ----
  	endif
  	## for JAVA
  	#setenv JAVA_HOME /usr/local/j2sdk1.4.0
! 	#setenv CLASSPATH ".:$JAVA_HOME/lib/tools.jar"
  	#if( $SHLVL == 1 ) then
  	#  set path = ( $path $JAVA_HOME/jre/bin $JAVA_HOME/bin )
  	#endif
diff -Nrc files/etc/npasswd files32/etc/npasswd
*** files/etc/npasswd	2013-11-11 17:27:08.000000000 +0900
--- files32/etc/npasswd	2013-11-11 17:27:08.000000000 +0900
***************
*** 7,13 ****
  shutdown:x:6:0:shutdown:/sbin:/sbin/shutdown
  halt:x:7:0:halt:/sbin:/sbin/halt
  mail:x:8:12:mail:/var/spool/mail:
! news:x:9:13:news:/usr/lib64/news:
  uucp:x:10:14:uucp:/var/spool/uucppublic:
  operator:x:11:0:operator:/root:/bin/bash
  games:x:12:100:games:/usr/games:
--- 7,13 ----
  shutdown:x:6:0:shutdown:/sbin:/sbin/shutdown
  halt:x:7:0:halt:/sbin:/sbin/halt
  mail:x:8:12:mail:/var/spool/mail:
! news:x:9:13:news:/usr/lib/news:
  uucp:x:10:14:uucp:/var/spool/uucppublic:
  operator:x:11:0:operator:/root:/bin/bash
  games:x:12:100:games:/usr/games:
***************
*** 15,21 ****
  postmaster:x:14:12:postmaster:/var/spool/mail:/bin/bash
  postfix:x:15:16:postfix:/dev/null:/dev/null
  pop:x:16:12:Pop Account:/dev/null:/bin/false
! wnn:x:17:1:Wnn:/usr/lib64/wnn:
  sshd:x:18:18:sshd privsev:/var/empty:/bin/false
  ftp:x:19:19:FTP server:/home/ftp:/bin/bash
  polkituser:x:21:21:PolicyKit daemon:/home/polkituser:/bin/true
--- 15,21 ----
  postmaster:x:14:12:postmaster:/var/spool/mail:/bin/bash
  postfix:x:15:16:postfix:/dev/null:/dev/null
  pop:x:16:12:Pop Account:/dev/null:/bin/false
! wnn:x:17:1:Wnn:/usr/lib/wnn:
  sshd:x:18:18:sshd privsev:/var/empty:/bin/false
  ftp:x:19:19:FTP server:/home/ftp:/bin/bash
  polkituser:x:21:21:PolicyKit daemon:/home/polkituser:/bin/true
diff -Nrc files/etc/profile files32/etc/profile
*** files/etc/profile	2009-08-07 20:49:14.000000000 +0900
--- files32/etc/profile	2009-08-07 20:49:14.000000000 +0900
***************
*** 12,18 ****
  TZ=/etc/localtime
  OUTPUT_CHARSET=EUC-JP 
  LANG=ja_JP.eucJP
! PKG_CONFIG_PATH=/usr/lib64/pkgconfig
  export MINICOM MANPATH HOSTNAME LESSOPEN
  export JLESSCHARSET LESS JSERVER TZ OUTPUT_CHARSET LANG PKG_CONFIG_PATH
  
--- 12,18 ----
  TZ=/etc/localtime
  OUTPUT_CHARSET=EUC-JP 
  LANG=ja_JP.eucJP
! PKG_CONFIG_PATH=/usr/lib/pkgconfig
  export MINICOM MANPATH HOSTNAME LESSOPEN
  export JLESSCHARSET LESS JSERVER TZ OUTPUT_CHARSET LANG PKG_CONFIG_PATH
  
***************
*** 37,43 ****
  ## for JAVA
  #JAVA_HOME=/usr/local/j2sdk1.4.0
  #export JAVA_HOME
! #CLASSPATH=.:$JAVA_HOME/lib64/tools.jar
  #export CLASSPATH
  #if [ $SHLVL = 1 ] ; then
  #  PATH="$PATH:$JAVA_HOME/jre/bin:$JAVA_HOME/bin"
--- 37,43 ----
  ## for JAVA
  #JAVA_HOME=/usr/local/j2sdk1.4.0
  #export JAVA_HOME
! #CLASSPATH=.:$JAVA_HOME/lib/tools.jar
  #export CLASSPATH
  #if [ $SHLVL = 1 ] ; then
  #  PATH="$PATH:$JAVA_HOME/jre/bin:$JAVA_HOME/bin"
diff -Nrc files/etc/template/Source/.bashrc files32/etc/template/Source/.bashrc
*** files/etc/template/Source/.bashrc	2016-01-22 17:19:56.000000000 +0900
--- files32/etc/template/Source/.bashrc	2016-01-22 17:19:56.000000000 +0900
***************
*** 53,59 ****
  export LESS EDITOR PAGER
  
  # for Python 2.7
! #PYTHONPATH=/usr/lib64/python2.7/site-packages:/opt/kde/lib64/python2.7/site-packages
  #export PYTHONPATH
  
  # if [ $SHLVL = 1 ] ; then
--- 53,59 ----
  export LESS EDITOR PAGER
  
  # for Python 2.7
! #PYTHONPATH=/usr/lib/python2.7/site-packages:/opt/kde/lib/python2.7/site-packages
  #export PYTHONPATH
  
  # if [ $SHLVL = 1 ] ; then
***************
*** 94,100 ****
  kterm|xterm|rxvt|dtterm|vt100|screen)
      # Window¤Î¥¿¥¤¥È¥ë¥Ð¡¼¤Ë¥«¥ì¥ó¥È¥Ç¥£¥ì¥¯¥È¥ê
      # ¤¬É½¼¨²ÄÇ½¤Ê¾ì¹ç¤Î¥×¥í¥ó¥×¥ÈÀßÄê
!     #   É½¼¨Îã¢ª hoge@host:/usr/lib64/X11/app-defaults$ _
      PS1='\u@\h:\w\$ '
      #   É½¼¨Îã¢ª hoge@host:app-defaults$ _
      #PS1='\u@\h:\W\$ '
--- 94,100 ----
  kterm|xterm|rxvt|dtterm|vt100|screen)
      # Window¤Î¥¿¥¤¥È¥ë¥Ð¡¼¤Ë¥«¥ì¥ó¥È¥Ç¥£¥ì¥¯¥È¥ê
      # ¤¬É½¼¨²ÄÇ½¤Ê¾ì¹ç¤Î¥×¥í¥ó¥×¥ÈÀßÄê
!     #   É½¼¨Îã¢ª hoge@host:/usr/lib/X11/app-defaults$ _
      PS1='\u@\h:\w\$ '
      #   É½¼¨Îã¢ª hoge@host:app-defaults$ _
      #PS1='\u@\h:\W\$ '
diff -Nrc files/etc/template/Source/.emacs.d/site-start.d/key_el files32/etc/template/Source/.emacs.d/site-start.d/key_el
*** files/etc/template/Source/.emacs.d/site-start.d/key_el	2011-08-31 15:18:54.000000000 +0900
--- files32/etc/template/Source/.emacs.d/site-start.d/key_el	2011-08-31 15:18:54.000000000 +0900
***************
*** 4,11 ****
  ;; ¤³¤³¤Ï¤è¤¯¤ï¤«¤ê¤Þ¤»¤Ì ^^;
  
  ;;; ¥Þ¥¯¥í¥µ¡¼¥Á¥Ñ¥¹¤ÎÄÉ²Ã
! ;;; ~/lib64/emacs °Ê²¼¤Ë¥æ¡¼¥¶ÍÑ¤Î *.el, *.elc ¤òÃÖ¤¯¤³¤È¤¬¤Ç¤­¤Þ¤¹
! ;;(setq load-path (append '("~/lib64/emacs") load-path))
  
  ;;; C-h ¤È Del ¤ÎÆþ¤ìÂØ¤¨
  ;;; Help ¤¬ Shift + Ctrl + h ¤ª¤è¤Ó Del ¤Ë³äÅö¤Æ¤é¤ì¡¢
--- 4,11 ----
  ;; ¤³¤³¤Ï¤è¤¯¤ï¤«¤ê¤Þ¤»¤Ì ^^;
  
  ;;; ¥Þ¥¯¥í¥µ¡¼¥Á¥Ñ¥¹¤ÎÄÉ²Ã
! ;;; ~/lib/emacs °Ê²¼¤Ë¥æ¡¼¥¶ÍÑ¤Î *.el, *.elc ¤òÃÖ¤¯¤³¤È¤¬¤Ç¤­¤Þ¤¹
! ;;(setq load-path (append '("~/lib/emacs") load-path))
  
  ;;; C-h ¤È Del ¤ÎÆþ¤ìÂØ¤¨
  ;;; Help ¤¬ Shift + Ctrl + h ¤ª¤è¤Ó Del ¤Ë³äÅö¤Æ¤é¤ì¡¢
diff -Nrc files/etc/template/Source/.steprc files32/etc/template/Source/.steprc
*** files/etc/template/Source/.steprc	2009-06-16 08:14:59.000000000 +0900
--- files32/etc/template/Source/.steprc	2009-06-16 08:14:59.000000000 +0900
***************
*** 170,176 ****
  # ModulePath is a colon-separated list, just like a regular unix PATH
  # Any Module must be found within the path in order to be located
  # and executed
! ModulePath          /usr/lib64/X11/afterstep
  # Another colon-separated list pointing to your pixmaps (color icons)
  PixmapPath          /usr/include/X11/pixmaps/:/usr/share/afterstep/icons:/usr/share/pixmaps
  # Path for your ugly black&white icons
--- 170,176 ----
  # ModulePath is a colon-separated list, just like a regular unix PATH
  # Any Module must be found within the path in order to be located
  # and executed
! ModulePath          /usr/lib/X11/afterstep
  # Another colon-separated list pointing to your pixmaps (color icons)
  PixmapPath          /usr/include/X11/pixmaps/:/usr/share/afterstep/icons:/usr/share/pixmaps
  # Path for your ugly black&white icons
diff -Nrc files/etc/template/Source/.tcshrc files32/etc/template/Source/.tcshrc
*** files/etc/template/Source/.tcshrc	2016-01-22 17:21:41.000000000 +0900
--- files32/etc/template/Source/.tcshrc	2016-01-22 17:21:41.000000000 +0900
***************
*** 43,49 ****
  setenv PAGER 'less'
  
  # for Python 2.7
! #setenv PYTHONPATH /usr/lib64/python2.7/site-packages:/opt/kde/lib64/python2.7/site-packages
  
  # NetNews ¤Î¥µ¡¼¥Ð¤ò»ØÄê¤·¤Æ¤¯¤À¤µ¤¤
  # setenv NNTPSERVER news.hoge.hoge.ac.jp
--- 43,49 ----
  setenv PAGER 'less'
  
  # for Python 2.7
! #setenv PYTHONPATH /usr/lib/python2.7/site-packages:/opt/kde/lib/python2.7/site-packages
  
  # NetNews ¤Î¥µ¡¼¥Ð¤ò»ØÄê¤·¤Æ¤¯¤À¤µ¤¤
  # setenv NNTPSERVER news.hoge.hoge.ac.jp
***************
*** 100,106 ****
  	endif
  	# Window¤Î¥¿¥¤¥È¥ë¥Ð¡¼¤Ë¥«¥ì¥ó¥È¥Ç¥£¥ì¥¯¥È¥ê
  	# ¤¬É½¼¨²ÄÇ½¤Ê¾ì¹ç¤Î¥×¥í¥ó¥×¥ÈÀßÄê
! 	#   É½¼¨Îã¢ª hoge@host[/usr/lib64/X11/app-defaults]% _
  	set prompt   = `whoami`@`hostname`"[%~]% "
  	#   É½¼¨Îã¢ª hoge@host[/<4>app-defaults]% _
  	#set prompt   = `whoami`@`hostname`"[%c01]% "
--- 100,106 ----
  	endif
  	# Window¤Î¥¿¥¤¥È¥ë¥Ð¡¼¤Ë¥«¥ì¥ó¥È¥Ç¥£¥ì¥¯¥È¥ê
  	# ¤¬É½¼¨²ÄÇ½¤Ê¾ì¹ç¤Î¥×¥í¥ó¥×¥ÈÀßÄê
! 	#   É½¼¨Îã¢ª hoge@host[/usr/lib/X11/app-defaults]% _
  	set prompt   = `whoami`@`hostname`"[%~]% "
  	#   É½¼¨Îã¢ª hoge@host[/<4>app-defaults]% _
  	#set prompt   = `whoami`@`hostname`"[%c01]% "
diff -Nrc files/etc/template/Source/.xinitrc files32/etc/template/Source/.xinitrc
*** files/etc/template/Source/.xinitrc	2016-01-22 18:36:39.000000000 +0900
--- files32/etc/template/Source/.xinitrc	2016-01-22 18:36:39.000000000 +0900
***************
*** 25,32 ****
    cur_mail=/var/spool/mail/$USER
    if [ -f $ini_mail -a -f $cur_mail ] ; then
      if diff $ini_mail $cur_mail > /dev/null ; then
!       [ -x /usr/lib64/X11/afterstep/plamo_banner ] &&
!           /usr/lib64/X11/afterstep/plamo_banner &
      fi
    fi
  elif [ $WM == "qvwm" ] ; then
--- 25,32 ----
    cur_mail=/var/spool/mail/$USER
    if [ -f $ini_mail -a -f $cur_mail ] ; then
      if diff $ini_mail $cur_mail > /dev/null ; then
!       [ -x /usr/lib/X11/afterstep/plamo_banner ] &&
!           /usr/lib/X11/afterstep/plamo_banner &
      fi
    fi
  elif [ $WM == "qvwm" ] ; then
***************
*** 44,50 ****
  
  # KDE additions:
  KDEDIR=/opt/kde
! PATH=$PATH:/usr/lib64/qt/bin:$KDEDIR/bin
  
  # Environment variables for the htdig package:
  PATH=$PATH:/opt/www/htdig/bin
--- 44,50 ----
  
  # KDE additions:
  KDEDIR=/opt/kde
! PATH=$PATH:/usr/lib/qt/bin:$KDEDIR/bin
  
  # Environment variables for the htdig package:
  PATH=$PATH:/opt/www/htdig/bin
***************
*** 71,77 ****
  #xmodmap -e 'keycode 131 = backslash underscore'
  
  # ¥æ¡¼¥¶¤Î¥Û¡¼¥à¥Ç¥£¥ì¥¯¥È¥ê¤Ë font ¤òÃÖ¤¯¾ì¹ç¤Ë¤Ï¤³¤Î¤è¤¦¤Ë¤·¤Þ¤¹¡£
! #xset +fp $HOME/lib64/fonts
  
  # ´Á»úÆþÎÏ¥·¥¹¥Æ¥à kinput2 ¤Îµ¯Æ°
  #CC_DEF_PATH=$HOME/.kinput2/ccdef
--- 71,77 ----
  #xmodmap -e 'keycode 131 = backslash underscore'
  
  # ¥æ¡¼¥¶¤Î¥Û¡¼¥à¥Ç¥£¥ì¥¯¥È¥ê¤Ë font ¤òÃÖ¤¯¾ì¹ç¤Ë¤Ï¤³¤Î¤è¤¦¤Ë¤·¤Þ¤¹¡£
! #xset +fp $HOME/lib/fonts
  
  # ´Á»úÆþÎÏ¥·¥¹¥Æ¥à kinput2 ¤Îµ¯Æ°
  #CC_DEF_PATH=$HOME/.kinput2/ccdef
***************
*** 129,136 ****
  # ATOKX ¥¯¥é¥¤¥¢¥ó¥È¤Îµ¯Æ°
  if [ -x /opt/atokx2/bin/atokx2_client.sh ] ; then
    /opt/atokx2/bin/atokx2_client.sh
! elif [ -x /usr/lib64/im/locale/ja/atokserver/atokx_client ] ; then
!   . /usr/lib64/im/locale/ja/atokserver/atokx_client
  fi
  
  # for StarSuite
--- 129,136 ----
  # ATOKX ¥¯¥é¥¤¥¢¥ó¥È¤Îµ¯Æ°
  if [ -x /opt/atokx2/bin/atokx2_client.sh ] ; then
    /opt/atokx2/bin/atokx2_client.sh
! elif [ -x /usr/lib/im/locale/ja/atokserver/atokx_client ] ; then
!   . /usr/lib/im/locale/ja/atokserver/atokx_client
  fi
  
  # for StarSuite
***************
*** 142,150 ****
  #export UNICODEMAP_JP
  
  # Environment variables for the Qt package:
! #QTDIR=/usr/lib64/qt
  #CPLUS_INCLUDE_PATH=$QTDIR/include:$CPLUS_INCLUDE_PATH
! #MANPATH="$MANPATH:/usr/lib64/qt/doc/man"
  #export QTDIR CPLUS_INCLUDE_PATH MANPATH
  # QT_GRAPHICSSYSTEM: native (default) | opengl | raster
  #QT_GRAPHICSSYSTEM=opengl
--- 142,150 ----
  #export UNICODEMAP_JP
  
  # Environment variables for the Qt package:
! #QTDIR=/usr/lib/qt
  #CPLUS_INCLUDE_PATH=$QTDIR/include:$CPLUS_INCLUDE_PATH
! #MANPATH="$MANPATH:/usr/lib/qt/doc/man"
  #export QTDIR CPLUS_INCLUDE_PATH MANPATH
  # QT_GRAPHICSSYSTEM: native (default) | opengl | raster
  #QT_GRAPHICSSYSTEM=opengl
***************
*** 174,180 ****
  #xv +noresetroot -root -max -quit /usr/share/pixmaps/backgrounds/Plamo/plamoback_c1.png
  #xv +noresetroot -root -max -quit /usr/share/pixmaps/backgrounds/Plamo/plamoback_c2.png
  #xv +noresetroot -root -max -quit /usr/share/pixmaps/backgrounds/Plamo/plamoback_c3.png
! #xv +noresetroot -root -quit /usr/lib64/X11/afterstep/backgrounds/wood.jpg
  #xv +noresetroot -root -quit /usr/include/X11/pixmaps/world.xpm
  #xv +noresetroot -root -quit /usr/share/pixmaps/backgrounds/Plamo/sx68k_root1.xpm
  #xv +noresetroot -root -quit /usr/share/pixmaps/backgrounds/Plamo/sx68k_root2.xpm
--- 174,180 ----
  #xv +noresetroot -root -max -quit /usr/share/pixmaps/backgrounds/Plamo/plamoback_c1.png
  #xv +noresetroot -root -max -quit /usr/share/pixmaps/backgrounds/Plamo/plamoback_c2.png
  #xv +noresetroot -root -max -quit /usr/share/pixmaps/backgrounds/Plamo/plamoback_c3.png
! #xv +noresetroot -root -quit /usr/lib/X11/afterstep/backgrounds/wood.jpg
  #xv +noresetroot -root -quit /usr/include/X11/pixmaps/world.xpm
  #xv +noresetroot -root -quit /usr/share/pixmaps/backgrounds/Plamo/sx68k_root1.xpm
  #xv +noresetroot -root -quit /usr/share/pixmaps/backgrounds/Plamo/sx68k_root2.xpm
diff -Nrc files/etc/template/Source/.zshrc files32/etc/template/Source/.zshrc
*** files/etc/template/Source/.zshrc	2016-01-22 19:52:16.000000000 +0900
--- files32/etc/template/Source/.zshrc	2016-01-22 19:52:16.000000000 +0900
***************
*** 40,46 ****
  export LESS EDITOR PAGER
  
  # for Python 2.7
! #PYTHONPATH=/usr/lib64/python2.7/site-packages:/opt/kde/lib64/python2.7/site-packages
  
  HISTFILE=$HOME/.zhistory
  
--- 40,46 ----
  export LESS EDITOR PAGER
  
  # for Python 2.7
! #PYTHONPATH=/usr/lib/python2.7/site-packages:/opt/kde/lib/python2.7/site-packages
  
  HISTFILE=$HOME/.zhistory
  
***************
*** 74,80 ****
  kterm|xterm|rxvt|dtterm|vt100)
      # Window¤Î¥¿¥¤¥È¥ë¥Ð¡¼¤Ë¥«¥ì¥ó¥È¥Ç¥£¥ì¥¯¥È¥ê
      # ¤¬É½¼¨²ÄÇ½¤Ê¾ì¹ç¤Î¥×¥í¥ó¥×¥ÈÀßÄê
!     #   É½¼¨Îã¢ª hoge@host[/usr/lib64/X11/app-defaults]% _
      PS1='%n@%m[%~]%# '
      #   É½¼¨Îã¢ª hoge@host[app-defaults]% _
      #PS1='%n@%m[%c]%# '
--- 74,80 ----
  kterm|xterm|rxvt|dtterm|vt100)
      # Window¤Î¥¿¥¤¥È¥ë¥Ð¡¼¤Ë¥«¥ì¥ó¥È¥Ç¥£¥ì¥¯¥È¥ê
      # ¤¬É½¼¨²ÄÇ½¤Ê¾ì¹ç¤Î¥×¥í¥ó¥×¥ÈÀßÄê
!     #   É½¼¨Îã¢ª hoge@host[/usr/lib/X11/app-defaults]% _
      PS1='%n@%m[%~]%# '
      #   É½¼¨Îã¢ª hoge@host[app-defaults]% _
      #PS1='%n@%m[%c]%# '
diff -Nrc files/etc/template/mime-mail.sample files32/etc/template/mime-mail.sample
*** files/etc/template/mime-mail.sample	2001-04-08 17:03:53.000000000 +0900
--- files32/etc/template/mime-mail.sample	2001-04-08 17:03:53.000000000 +0900
***************
*** 54,63 ****
    $B$j$^$9!#(B/etc/XF86Config $B$NNc$r<($7$^$9!#(B
  
  Section "Files"
!    RgbPath    "/usr/lib64/X11/rgb"
     FontPath   "tcp/localhost:7100"
!    FontPath   "/usr/lib64/X11/fonts/misc:unscaled"
!    FontPath   "/usr/lib64/X11/fonts/75dpi:unscaled"
  
  $B!y%;%-%e%j%F%#$N4QE@$+$i!"(B/etc/hosts.deny $B$H(B /etc/hosts.allow $B$K!"(B
    ALL : ALL $B!"(B ALL : LOCAL $B$H$$$&@_Dj$K$7$F$$$^$9!#$=$N$?$a!"%G%U%)%k(B
--- 54,63 ----
    $B$j$^$9!#(B/etc/XF86Config $B$NNc$r<($7$^$9!#(B
  
  Section "Files"
!    RgbPath    "/usr/lib/X11/rgb"
     FontPath   "tcp/localhost:7100"
!    FontPath   "/usr/lib/X11/fonts/misc:unscaled"
!    FontPath   "/usr/lib/X11/fonts/75dpi:unscaled"
  
  $B!y%;%-%e%j%F%#$N4QE@$+$i!"(B/etc/hosts.deny $B$H(B /etc/hosts.allow $B$K!"(B
    ALL : ALL $B!"(B ALL : LOCAL $B$H$$$&@_Dj$K$7$F$$$^$9!#$=$N$?$a!"%G%U%)%k(B
