--- mozilla-1.9.2/modules/libpr0n/encoders/png/nsPNGEncoder.cpp.orig	2011-01-14 17:47:57.000000000 +0900
+++ mozilla-1.9.2/modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2011-01-14 17:49:02.000000000 +0900
@@ -135,9 +135,13 @@
 
   // initialize
   mPNG = png_create_write_struct(PNG_LIBPNG_VER_STRING,
+#if PNG_LIBPNG_VER < 10400
                                  png_voidp_NULL,
                                  ErrorCallback,
                                  ErrorCallback);
+#else
+                                 nsnull,nsnull,nsnull);
+#endif
   if (! mPNG)
     return NS_ERROR_OUT_OF_MEMORY;
 
