diff -ur emacs-24.4.orig/lisp/eshell/esh-util.el emacs-24.4/lisp/eshell/esh-util.el
--- emacs-24.4.orig/lisp/eshell/esh-util.el	2014-03-21 14:34:40.000000000 +0900
+++ emacs-24.4/lisp/eshell/esh-util.el	2014-10-29 15:45:20.878182611 +0900
@@ -72,7 +72,7 @@
   :group 'eshell-util)
 
 (defcustom eshell-tar-regexp
-  "\\.t\\(ar\\(\\.\\(gz\\|bz2\\|xz\\|Z\\)\\)?\\|gz\\|a[zZ]\\|z2\\)\\'"
+  "\\.t\\(ar\\(\\.\\(gz\\|bz2\\|xz\\|Z\\)\\)?\\|gz\\|a[zZ]\\|z2\\|xz\\)\\'"
   "Regular expression used to match tar file names."
   :version "24.1"			; added xz
   :type 'regexp
diff -ur emacs-24.4.orig/lisp/ibuffer.el emacs-24.4/lisp/ibuffer.el
--- emacs-24.4.orig/lisp/ibuffer.el	2014-03-21 14:34:40.000000000 +0900
+++ emacs-24.4/lisp/ibuffer.el	2014-10-29 15:46:05.650334998 +0900
@@ -332,7 +332,7 @@
   :group 'ibuffer)
 
 (defcustom ibuffer-compressed-file-name-regexp
-  "\\.\\(arj\\|bgz\\|bz2\\|gz\\|lzh\\|taz\\|tgz\\|xz\\|zip\\|z\\)$"
+  "\\.\\(arj\\|bgz\\|bz2\\|gz\\|lzh\\|taz\\|tgz\\|xz\\|zip\\|z\\|txz\\)$"
   "Regexp to match compressed file names."
   :version "24.1"                       ; added xz
   :type 'regexp
diff -ur emacs-24.4.orig/lisp/ldefs-boot.el emacs-24.4/lisp/ldefs-boot.el
--- emacs-24.4.orig/lisp/ldefs-boot.el	2014-10-17 13:51:02.000000000 +0900
+++ emacs-24.4/lisp/ldefs-boot.el	2014-10-29 15:46:35.475102159 +0900
@@ -9366,7 +9366,7 @@
 
 (custom-autoload 'tags-table-list "etags" t)
 
-(defvar tags-compression-info-list (purecopy '("" ".Z" ".bz2" ".gz" ".xz" ".tgz")) "\
+(defvar tags-compression-info-list (purecopy '("" ".Z" ".bz2" ".gz" ".xz" ".tgz" ".txz")) "\
 List of extensions tried by etags when `auto-compression-mode' is on.
 An empty string means search the non-compressed file.")
 
diff -ur emacs-24.4.orig/lisp/loaddefs.el emacs-24.4/lisp/loaddefs.el
--- emacs-24.4.orig/lisp/loaddefs.el	2014-10-18 00:57:21.000000000 +0900
+++ emacs-24.4/lisp/loaddefs.el	2014-10-29 15:46:55.549400577 +0900
@@ -9366,7 +9366,7 @@
 
 (custom-autoload 'tags-table-list "etags" t)
 
-(defvar tags-compression-info-list (purecopy '("" ".Z" ".bz2" ".gz" ".xz" ".tgz")) "\
+(defvar tags-compression-info-list (purecopy '("" ".Z" ".bz2" ".gz" ".xz" ".tgz" ".txz")) "\
 List of extensions tried by etags when `auto-compression-mode' is on.
 An empty string means search the non-compressed file.")
 
diff -ur emacs-24.4.orig/lisp/pcmpl-gnu.el emacs-24.4/lisp/pcmpl-gnu.el
--- emacs-24.4.orig/lisp/pcmpl-gnu.el	2014-03-21 14:34:40.000000000 +0900
+++ emacs-24.4/lisp/pcmpl-gnu.el	2014-10-29 15:48:02.059264808 +0900
@@ -90,7 +90,7 @@
     (lambda (entry)
       (when (and (file-readable-p entry)
 		 (file-regular-p entry))
-	(let ((zipped (string-match "\\.\\(t?z2\\|bz2\\)\\'" entry)))
+	(let ((zipped (string-match "\\.\\(t?z2\\|bz2\\|txz\\)\\'" entry)))
 	  (or (and unzip-p zipped)
 	      (and (not unzip-p) (not zipped)))))))))
 
@@ -128,7 +128,7 @@
       (pcomplete-uniqify-list rules))))
 
 (defcustom pcmpl-gnu-tarfile-regexp
-  "\\.t\\(ar\\(\\.\\(gz\\|bz2\\|Z\\|xz\\)\\)?\\|gz\\|a[zZ]\\|z2\\)\\'"
+  "\\.t\\(ar\\(\\.\\(gz\\|bz2\\|Z\\|xz\\)\\)?\\|gz\\|a[zZ]\\|z2\\|xz\\)\\'"
   "A regexp which matches any tar archive."
   :version "24.3"                       ; added xz
   :type 'regexp
