*** build/version.h~	Sat Sep 25 00:00:00 2010
--- build/version.h	Sun Nov  7 09:57:13 2010
***************
*** 4,8 ****
   *	version number
   */
  
! char *distributor = NULL;
  static char version[] = "@(#)fd.c  3.00j 09/25/10";
--- 4,9 ----
   *	version number
   */
  
! /* char *distributor = NULL; */
! char *distributor = "kojima@linet.gr.jp (Plamo Linux)";
  static char version[] = "@(#)fd.c  3.00j 09/25/10";
