diff -upr LVM2.2.02.90/configure.in LVM2.2.02.90.patched/configure.in
--- LVM2.2.02.90/configure.in	2012-02-01 06:21:54.000000000 +0900
+++ LVM2.2.02.90.patched/configure.in	2012-11-05 14:18:12.062542440 +0900
@@ -32,6 +32,7 @@ case "$host_os" in
 		COPTIMISE_FLAG="-O2"
 		CLDFLAGS="$CLDFLAGS -Wl,--version-script,.export.sym"
 		ELDFLAGS="-Wl,--export-dynamic"
+		STATIC_LDFLAGS="-Wl,--no-export-dynamic"
 		# FIXME Generate list and use --dynamic-list=.dlopen.sym
 		CLDWHOLEARCHIVE="-Wl,-whole-archive"
 		CLDNOWHOLEARCHIVE="-Wl,-no-whole-archive"
@@ -1408,6 +1409,7 @@ AC_SUBST(SELINUX_LIBS)
 AC_SUBST(SELINUX_PC)
 AC_SUBST(SNAPSHOTS)
 AC_SUBST(STATICDIR)
+AC_SUBST(STATIC_LDFLAGS)
 AC_SUBST(STATIC_LINK)
 AC_SUBST(TESTING)
 AC_SUBST(THIN)
LVM2.2.02.90/daemons/dmeventdだけに発見: .#Makefile.in
diff -upr LVM2.2.02.90/daemons/dmeventd/Makefile.in LVM2.2.02.90.patched/daemons/dmeventd/Makefile.in
--- LVM2.2.02.90/daemons/dmeventd/Makefile.in	2011-11-15 06:30:36.000000000 +0900
+++ LVM2.2.02.90.patched/daemons/dmeventd/Makefile.in	2012-11-05 14:19:13.762754466 +0900
@@ -64,7 +64,7 @@ dmeventd: $(LIB_SHARED) dmeventd.o
 	$(DL_LIBS) $(LVMLIBS) $(LIBS) -rdynamic
 
 dmeventd.static: $(LIB_STATIC) dmeventd.o $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) $(ELDFLAGS) -static -L. -L$(interfacebuilddir) -o $@ \
+	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L. -L$(interfacebuilddir) -o $@ \
 	dmeventd.o $(DL_LIBS) $(LVMLIBS) $(LIBS) $(STATIC_LIBS)
 
 ifeq ("@PKGCONFIG@", "yes")
diff -upr LVM2.2.02.90/tools/Makefile.in LVM2.2.02.90.patched/tools/Makefile.in
--- LVM2.2.02.90/tools/Makefile.in	2011-11-15 06:30:36.000000000 +0900
+++ LVM2.2.02.90.patched/tools/Makefile.in	2012-11-05 14:17:27.742389862 +0900
@@ -126,7 +126,7 @@ dmsetup: dmsetup.o $(top_builddir)/libdm
 	      -o $@ dmsetup.o -ldevmapper $(LIBS)
 
 dmsetup.static: dmsetup.o $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) \
+	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L$(interfacebuilddir) \
 	      -o $@ dmsetup.o -ldevmapper $(STATIC_LIBS) $(LIBS)
 
 all: device-mapper
@@ -136,7 +136,7 @@ lvm: $(OBJECTS) lvm.o $(top_builddir)/li
 		$(LVMLIBS) $(READLINE_LIBS) $(LIBS) -rdynamic
 
 lvm.static: $(OBJECTS) lvm-static.o $(top_builddir)/lib/liblvm-internal.a  $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) -o $@ \
+	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L$(interfacebuilddir) -o $@ \
 	      $(OBJECTS) lvm-static.o $(LVMLIBS) $(STATIC_LIBS) $(LIBS)
 
 liblvm2cmd.a: $(top_builddir)/lib/liblvm-internal.a $(OBJECTS) lvmcmdlib.o lvm2cmd.o
