diff -uNr libxfce4ui-4.10.0.orig/libxfce4ui/xfce-dialogs.c libxfce4ui-4.10.0/libxfce4ui/xfce-dialogs.c
--- libxfce4ui-4.10.0.orig/libxfce4ui/xfce-dialogs.c	2012-04-29 04:46:52.000000000 +0900
+++ libxfce4ui-4.10.0/libxfce4ui/xfce-dialogs.c	2013-01-08 12:55:11.076146079 +0900
@@ -144,6 +144,7 @@
   GdkScreen   *screen;
 
   g_return_if_fail (parent == NULL || GTK_IS_WINDOW (parent));
+  xfce_textdomain(GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR, "UTF-8");
 
   /* get the user's locale without encoding */
   locale = g_strdup (setlocale (LC_MESSAGES, NULL));
@@ -661,7 +662,7 @@
   g_return_val_if_fail (parent == NULL || GTK_IS_WINDOW (parent), 0);
 
   va_start (args, first_button_text);
-  dialog = xfce_message_dialog_new_valist (parent, title, stock_id, primary_text,
+  dialog = xfce_message_dialog_new_valist (parent, g_filename_display_name(title), stock_id, primary_text,
                                            secondary_text, first_button_text, args);
   va_end (args);
 
