diff -uNr openssl-1.1.0h/tools/c_rehash.in openssl-1.1.0h.new/tools/c_rehash.in
--- openssl-1.1.0h/tools/c_rehash.in	2018-03-27 22:50:41.000000000 +0900
+++ openssl-1.1.0h.new/tools/c_rehash.in	2018-05-09 15:26:59.882710480 +0900
@@ -11,8 +11,8 @@
 # Perl c_rehash script, scan all files in a directory
 # and add symbolic links to their hash values.
 
-my $dir = {- quotify1($config{openssldir}) -};
-my $prefix = {- quotify1($config{prefix}) -};
+my $dir = "{- quotify1($config{openssldir}) -}";
+my $prefix = "{- quotify1($config{prefix}) -}";
 
 my $errorcount = 0;
 my $openssl = $ENV{OPENSSL} || "openssl";
