--- libtta-c-2.0/Makefile.in	2010-12-30 21:44:12.000000000 +0900
+++ build/Makefile.in	2011-01-29 14:28:40.000000000 +0900
@@ -110,7 +110,7 @@
 CCAS = @CCAS@
 CCASFLAGS = @CCASFLAGS@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -fPIC
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CPU_ARM_FALSE = @CPU_ARM_FALSE@
@@ -194,7 +194,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 AUTOMAKE_OPTIONS = no-dependencies foreign
-lib_LIBRARIES = libtta-c.a
+lib_LIBRARIES = libtta-c.a libtta-c.so.2.0
 libtta_c_a_SOURCES = libtta.c libtta.h filter.h
 @CPU_ARM_TRUE@@ENABLE_ASM_TRUE@EXTRA_libtta_c_a_SOURCES = filter_arm.S
 @CPU_ARM_TRUE@@ENABLE_ASM_TRUE@libtta_c_a_LIBADD = filter_arm.o
@@ -267,7 +267,7 @@
 	  else :; fi; \
 	done
 	@$(POST_INSTALL)
-	@list='$(lib_LIBRARIES)'; for p in $$list; do \
+	@list='libtta-c.a'; for p in $$list; do \
 	  if test -f $$p; then \
 	    p=$(am__strip_dir) \
 	    echo " $(RANLIB) '$(DESTDIR)$(libdir)/$$p'"; \
@@ -290,6 +290,10 @@
 	$(libtta_c_a_AR) libtta-c.a $(libtta_c_a_OBJECTS) $(libtta_c_a_LIBADD)
 	$(RANLIB) libtta-c.a
 
+libtta-c.so.2.0:	$(libtta_c_a_OBJECTS) $(libtta_c_a_DEPENDENCIES) 
+	-rm -f libtta-c.so.2.0
+	$(CC) -o $@ -shared $?
+
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
