diff -Nru a/libavcodec/vp3.c b/libavcodec/vp3.c
--- a/libavcodec/vp3.c	2025-12-05 22:54:54.969284396 +0900
+++ b/libavcodec/vp3.c	2025-12-05 22:51:27.893385270 +0900
@@ -46,7 +46,6 @@
 #include "decode.h"
 #include "get_bits.h"
 #include "hpeldsp.h"
-#include "internal.h"
 #include "jpegquanttables.h"
 #include "mathops.h"
 #include "progressframe.h"
@@ -2458,7 +2457,7 @@
         }
     }
 
-    if (!avctx->internal->is_copy) {
+    if (ff_thread_sync_ref(avctx, offsetof(Vp3DecodeContext, coeff_vlc)) != FF_THREAD_IS_COPY) {
         CoeffVLCs *vlcs = ff_refstruct_alloc_ext(sizeof(*s->coeff_vlc), 0,
                                                  NULL, free_vlc_tables);
         if (!vlcs)
@@ -2527,8 +2526,6 @@
     const Vp3DecodeContext *s1 = src->priv_data;
     int qps_changed = 0;
 
-    ff_refstruct_replace(&s->coeff_vlc, s1->coeff_vlc);
-
     // copy previous frame data
     ref_frames(s, s1);
     if (!s1->current_frame.f ||
