diff --git a/nss/Makefile b/nss/Makefile
index eb4ed1a..cd7b08f 100644
--- a/nss/Makefile
+++ b/nss/Makefile
@@ -4,6 +4,8 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+default: nss_build_all
+
 #######################################################################
 # (1) Include initial platform-independent assignments (MANDATORY).   #
 #######################################################################
@@ -48,12 +50,10 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 #######################################################################
 
 nss_build_all:
-	$(MAKE) build_nspr
 	$(MAKE) all
 	$(MAKE) latest
 
 nss_clean_all:
-	$(MAKE) clobber_nspr
 	$(MAKE) clobber
 
 NSPR_CONFIG_STATUS = $(CORE_DEPTH)/../nspr/$(OBJDIR_NAME)/config.status
diff --git a/nss/manifest.mn b/nss/manifest.mn
index fbc420a..b983d88 100644
--- a/nss/manifest.mn
+++ b/nss/manifest.mn
@@ -10,7 +10,7 @@ IMPORTS =	nspr20/v4.8 \
 
 RELEASE = nss
 
-DIRS = coreconf lib cmd cpputil gtests
+DIRS = coreconf lib cmd cpputil gtests config
 
 HAVE_ALL_TARGET := 1
 
