diff -up source/texk/web2c/mplibdir/mp.w.externfix source/texk/web2c/mplibdir/mp.w
--- source/texk/web2c/mplibdir/mp.w.externfix	2020-01-23 13:46:37.500459473 -0500
+++ source/texk/web2c/mplibdir/mp.w	2020-01-23 13:55:29.588893404 -0500
@@ -2958,7 +2958,7 @@ void *mp_xmalloc (MP mp, size_t nmem, si
 }
 
 @ @<Internal library declarations@>=
-int mp_snprintf_res ;
+static int mp_snprintf_res ;
 /* Some compilers (i.e. gcc 8.2.0 ) complained with the old */
 /* #define mp_snprintf (void)snprintf                       */
 /* about truncation. For the moment we store the result.    */
