diff -uNr lxc-0.9.0.orig/config/default.conf.unknown lxc-0.9.0/config/default.conf.unknown
--- lxc-0.9.0.orig/config/default.conf.unknown	2013-02-18 16:47:43.000000000 +0900
+++ lxc-0.9.0/config/default.conf.unknown	2013-04-26 18:23:16.877274323 +0900
@@ -1 +1,6 @@
+# If you use /etc/rc.d/init.d/lxc-net, enable following 3 lines.
+#lxc.network.type=veth 
+#lxc.network.link=lxcbr0
+#lxc.network.flags=up
+# else if you make configure yourself, make appropriate changes for following lines
 lxc.network.type = empty
